### DISCUSSION TITLE
Pip instal web3 isn't working

### DISCUSSION THREAD

**Discussion by karthikvarma22** [smartcontractkit/full-blockchain-solidity-course-py#120] [Q&A] [answered]
pip3 install web3
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
Collecting web3
  Using cached web3-5.24.0-py3-none-any.whl (487 kB)
Requirement already satisfied: eth-typing<3.0.0,>=2.0.0 in /usr/local/lib/python3.9/site-packages (from web3) (2.2.2)
Requirement already satisfied: eth-abi<3.0.0,>=2.0.0b6 in /usr/local/lib/python3.9/site-packages (from web3) (2.1.1)
Collecting ipfshttpclient==0.8.0a2
  Using cached ipfshttpclient-0.8.0a2-py3-none-any.whl (82 kB)
Requirement already satisfied: requests<3.0.0,>=2.16.0 in /usr/local/lib/python3.9/site-packages (from web3) (2.26.0)
Collecting jsonschema<4.0.0,>=3.2.0
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting eth-account<0.6.0,>=0.5.6
  Using cached eth_account-0.5.6-py3-none-any.whl (101 kB)
Collecting aiohttp<4,>=3.7.4.post0
  Using cached aiohttp-3.7.4.post0-cp39-cp39-macosx_10_14_x86_64.whl (649 kB)
Requirement already satisfied: eth-hash[pycryptodome]<1.0.0,>=0.2.0 in /usr/local/lib/python3.9/site-packages (from web3) (0.3.2)
Requirement already satisfied: hexbytes<1.0.0,>=0.1.0 in /usr/local/lib/python3.9/site-packages (from web3) (0.2.2)
Collecting protobuf<4,>=3.10.0
  Using cached protobuf-3.18.0-cp39-cp39-macosx_10_9_x86_64.whl (1.0 MB)
Collecting websockets<10,>=9.1
  Using cached websockets-9.1-cp39-cp39-macosx_10_9_x86_64.whl (88 kB)
Requirement already satisfied: eth-utils<2.0.0,>=1.9.5 in /usr/local/lib/python3.9/site-packages (from web3) (1.10.0)
Collecting lru-dict<2.0.0,>=1.1.6
  Using cached lru-dict-1.1.7.tar.gz (10 kB)
Requirement already satisfied: multiaddr>=0.0.7 in /usr/local/lib/python3.9/site-packages (from ipfshttpclient==0.8.0a2->web3) (0.0.9)
Collecting attrs>=17.3.0
  Using cached attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting async-timeout<4.0,>=3.0
  Using cached async_timeout-3.0.1-py3-none-any.whl (8.2 kB)
Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.9/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (5.1.0)
Requirement already satisfied: typing-extensions>=3.6.5 in /usr/local/lib/python3.9/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (3.10.0.2)
Requirement already satisfied: chardet<5.0,>=2.0 in /usr/local/lib/python3.9/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (4.0.0)
Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.9/site-packages (from aiohttp<4,>=3.7.4.post0->web3) (1.6.3)
Requirement already satisfied: parsimonious<0.9.0,>=0.8.0 in /usr/local/lib/python3.9/site-packages (from eth-abi<3.0.0,>=2.0.0b6->web3) (0.8.1)
Requirement already satisfied: eth-rlp<2,>=0.1.2 in /usr/local/lib/python3.9/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.2.1)
Requirement already satisfied: eth-keyfile<0.6.0,>=0.5.0 in /usr/local/lib/python3.9/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.5.1)
Collecting bitarray<1.3.0,>=1.2.1
  Using cached bitarray-1.2.2.tar.gz (48 kB)
Requirement already satisfied: rlp<3,>=1.0.0 in /usr/local/lib/python3.9/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (2.0.1)
Requirement already satisfied: eth-keys!=0.3.2,<0.4.0,>=0.2.1 in /usr/local/lib/python3.9/site-packages (from eth-account<0.6.0,>=0.5.6->web3) (0.3.3)
Requirement already satisfied: pycryptodome<4,>=3.6.6 in /usr/local/lib/python3.9/site-packages (from eth-hash[pycryptodome]<1.0.0,>=0.2.0->web3) (3.10.4)
Requirement already satisfied: cytoolz<1.0.0,>=0.9.0 in /usr/local/lib/python3.9/site-packages (from eth-keyfile<0.6.0,>=0.5.0->eth-account<0.6.0,>=0.5.6->web3) (0.11.0)
Requirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.9/site-packages (from cytoolz<1.0.0,>=0.9.0->eth-keyfile<0.6.0,>=0.5.0->eth-account<0.6.0,>=0.5.6->web3) (0.11.1)
Requirement already satisfied: setuptools in /usr/local/lib/python3.9/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (57.4.0)
Requirement already satisfied: pyrsistent>=0.14.0 in /usr/local/lib/python3.9/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (0.18.0)
Requirement already satisfied: six>=1.11.0 in /usr/local/lib/python3.9/site-packages (from jsonschema<4.0.0,>=3.2.0->web3) (1.16.0)
Requirement already satisfied: base58 in /usr/local/lib/python3.9/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (2.1.0)
Requirement already satisfied: netaddr in /usr/local/lib/python3.9/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (0.8.0)
Requirement already satisfied: varint in /usr/local/lib/python3.9/site-packages (from multiaddr>=0.0.7->ipfshttpclient==0.8.0a2->web3) (1.0.2)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0,>=2.16.0->web3) (3.2)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0,>=2.16.0->web3) (2021.5.30)
Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0,>=2.16.0->web3) (2.0.6)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/site-packages (from requests<3.0.0,>=2.16.0->web3) (1.26.7)
Building wheels for collected packages: bitarray, lru-dict
  Building wheel for bitarray (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/opt/python@3.9/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"'; __file__='"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-wheel-bhbraoi0
       cwd: /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/
  Complete output (17 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-11-x86_64-3.9
  creating build/lib.macosx-11-x86_64-3.9/bitarray
  copying bitarray/util.py -> build/lib.macosx-11-x86_64-3.9/bitarray
  copying bitarray/__init__.py -> build/lib.macosx-11-x86_64-3.9/bitarray
  copying bitarray/test_bitarray.py -> build/lib.macosx-11-x86_64-3.9/bitarray
  copying bitarray/test_util.py -> build/lib.macosx-11-x86_64-3.9/bitarray
  running build_ext
  building 'bitarray._bitarray' extension
  creating build/temp.macosx-11-x86_64-3.9
  creating build/temp.macosx-11-x86_64-3.9/bitarray
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c bitarray/_bitarray.c -o build/temp.macosx-11-x86_64-3.9/bitarray/_bitarray.o
  clang: error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
  error: command '/usr/bin/clang' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for bitarray
  Running setup.py clean for bitarray
  Building wheel for lru-dict (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/opt/python@3.9/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/lru-dict_f24ef7c59f804be79c2f40094e230b28/setup.py'"'"'; __file__='"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/lru-dict_f24ef7c59f804be79c2f40094e230b28/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-wheel-i_mcn2xy
       cwd: /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/lru-dict_f24ef7c59f804be79c2f40094e230b28/
  Complete output (9 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'lru' extension
  creating build
  creating build/temp.macosx-11-x86_64-3.9
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c lru.c -o build/temp.macosx-11-x86_64-3.9/lru.o
  clang: error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
  error: command '/usr/bin/clang' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for lru-dict
  Running setup.py clean for lru-dict
Failed to build bitarray lru-dict
Installing collected packages: bitarray, attrs, async-timeout, websockets, protobuf, lru-dict, jsonschema, ipfshttpclient, eth-account, aiohttp, web3
    Running setup.py install for bitarray ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/local/opt/python@3.9/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"'; __file__='"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-record-2xhwsbuc/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.9/bitarray
         cwd: /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/
    Complete output (17 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-11-x86_64-3.9
    creating build/lib.macosx-11-x86_64-3.9/bitarray
    copying bitarray/util.py -> build/lib.macosx-11-x86_64-3.9/bitarray
    copying bitarray/__init__.py -> build/lib.macosx-11-x86_64-3.9/bitarray
    copying bitarray/test_bitarray.py -> build/lib.macosx-11-x86_64-3.9/bitarray
    copying bitarray/test_util.py -> build/lib.macosx-11-x86_64-3.9/bitarray
    running build_ext
    building 'bitarray._bitarray' extension
    creating build/temp.macosx-11-x86_64-3.9
    creating build/temp.macosx-11-x86_64-3.9/bitarray
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c bitarray/_bitarray.c -o build/temp.macosx-11-x86_64-3.9/bitarray/_bitarray.o
    clang: error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11'
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/opt/python@3.9/bin/python3.9 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"'; __file__='"'"'/private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-install-yoi8ebze/bitarray_d48753958aa045479c919d3de7535713/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/hy/hgwd99y91fbc2fmb6kxjh2xm0000gn/T/pip-record-2xhwsbuc/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.9/bitarray Check the logs for full command output.

This is the error that I am getting when i am trying pip3 install web3 command on my macOS. Any solutions to this ?

**Comment by karthikvarma22** [comment, marked as answer]
Worked
Had to use the following commands and then use pip3 install web3
sudo rm -rf /Library/Developer/CommandLineTools
sudo xcode-select --install

**Reply by AlbertoSinigaglia** [reply]
for the future discussion, please format your questiona and answer, remember that you can paste code using back-ticks to be highlighted:
````python
```python
from X import Y
```
````

**Reply by PatrickAlphaC** [reply]
Thanks @AlbertoSinigaglia 

Thanks for following up @joker-22

**Comment by 350z195** [comment]
This had me stuck for days. I figured it was either a problem with python or vscode, but it was actually a C compiler issue that is resolved by installing vsbuild tools. I came to this post and scrolled down to the second answer, and it fixed everything :). This is what they suggested: 

You need windows 10 SDK, Download visual studio build tools and install

Visual C++ Build tools core features.
MSVC toolset C++ 2019 v142 (x86,x64)
Visual C++ 2019 Redistributable Update
Windows 10 SDK (10.0.17763.0) for Desktop C++

orginal post:https://stackoverflow.com/questions/40018405/cannot-open-include-file-io-h-no-such-file-or-directory/50210015#50210015

Hope it helps!

**Reply by PatrickAlphaC** [reply]
Boom! Thanks for posting here.
